catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -109,6 +132,75 @@ public void init() {
           res.header("Content-Encoding", "gzip");
         });
 
+    post(
+        "/cql/transform/:transformerId",
+        (req, res) -> {
+          String transformerId = req.params(":transformerId");
+          String body = util.safeGetBody(req);
+          CqlRequest cqlRequest = mapper.readValue(body, CqlRequest.class);
+          CqlQueryResponse cqlQueryResponse = executeCqlQuery(cqlRequest);
+
+          for (ServiceReference<QueryResponseTransformer> queryResponseTransformer :
+              queryResponseTransformers) {
+
+            LOGGER.trace("Finding transformer to transform query response.");
+
+            String id = (String) queryResponseTransformer.getProperty("id");
+
+            if (id.equals(transformerId)) {
+              BinaryContent content =
+                  bundleContext
+                      .getService(queryResponseTransformer)
+                      .transform(cqlQueryResponse.getQueryResponse(), new HashMap<>());
+
+              String mimeType = (String) queryResponseTransformer.getProperty("mime-type");
+              MimeTypes allTypes = MimeTypes.getDefaultMimeTypes();
+              String fileExt = allTypes.forName(mimeType).getExtension();
+
+              res.status(200);
+
+              String acceptEncoding = req.headers("Accept-Encoding");
+
+              boolean shouldGzip =
+                  StringUtils.isNotBlank(acceptEncoding)
+                      && acceptEncoding.toLowerCase().contains("gzip");
+
+              if (!shouldGzip) {
+                LOGGER.trace("Request header does not contain gzip.");
+              }
+
+              res.type(mimeType);
+              String attachment =
+                  String.format(
+                      "attachment;filename=export-%s%s", Instant.now().toString(), fileExt);
+              res.header("Content-Disposition", attachment);
+
+              if (shouldGzip) {
+                res.raw().addHeader("Content-Encoding", "gzip");
+              }
+
+              try (OutputStream servletOutputStream = res.raw().getOutputStream();
+                  InputStream resultStream = content.getInputStream()) {
+                if (shouldGzip) {
```

Comment:
According to the [Spark documentation](http://sparkjava.com/documentation#gzip), to GZIP the response you only have to add the header and it will be done automatically.

Reply:
Removing the GZIP OutputStream (leaving just what's in the else block) causes an net::ERR_CONTENT_DECODING_FAILED response